python/Ganga/Core/GangaRepository/SubJobXMLList.py

```diff
@@ -164,15 +169,24 @@ def __really_writeIndex(self):
 
         raw_self = stripProxy(self)
 
+        import os
+
         all_caches = {}
-        for i in range(len(raw_self)):
-            this_cache = raw_self._registry.getIndexCache( stripProxy(raw_self.__getitem__(i)) )
-            all_caches[i] = this_cache
-            disk_location = raw_self.__get_dataFile(i)
-            import os
-            all_caches[i]['modified'] = os.stat(disk_location).st_ctime
+        for sj_id in range(len(raw_self)):
```

Comment:
No, I imagine that it does indeed call `__getitem__`. I would suggest at least a comment on this bit of code to explain the unusual idiom.